demisto_sdk/commands/pre_commit/pre_commit_command.py

```diff
@@ -0,0 +1,311 @@
+PRECOMMIT_PATH = CONTENT_PATH / ".pre-commit-config.yaml"
+
+SKIPPED_HOOKS = {"format", "validate"}
+
+INTEGRATION_SCRIPT_REGEX = re.compile(r"^Packs/.*/(?:Integrations|Scripts)/.*.yml$")
+
+# change to the latest demisto-sdk commit hash to debug
+DEMISTO_SDK_COMMIT_HASH_DEBUG = "c438943116c45cb67e6725220e7d5deaed73b802"
+
+
+@dataclass
+class PreCommitRunner:
+    """This class is responsible of running pre-commit hooks."""
+
+    python_version_to_files: Dict[str, Set[Path]]
+
+    def __post_init__(self):
+        """
+        We initialize the hooks and all_files for later use.
+        """
+        self.all_files = set(
+            itertools.chain.from_iterable(self.python_version_to_files.values())
+        )
+        with open(PRECOMMIT_TEMPLATE_PATH) as f:
+            self.precommit_template = yaml.load(f)
+
+        # changes the demisto-sdk revision to the latest release version (or the debug commit hash)
+        # to debug, modify the DEMISTO_SDK_COMMIT_HASH_DEBUG variable to your demisto-sdk commit hash
+        self.repos(self.precommit_template)["https://github.com/demisto/demisto-sdk"][
+            "rev"
+        ] = (DEMISTO_SDK_COMMIT_HASH_DEBUG or f"v{get_last_remote_release_version()}")
+
+    @staticmethod
+    def repos(pre_commit_config: dict) -> dict:
+        repos = {}
+        for repo in pre_commit_config["repos"]:
+            repos[repo["repo"]] = repo
+        return repos
+
+    @staticmethod
+    def hooks(pre_commit_config: dict) -> dict:
+        hooks = {}
+        for repo in pre_commit_config["repos"]:
+            for hook in repo["hooks"]:
+                hooks[hook["id"]] = hook
+        return hooks
+
+    def prepare_hooks(
+        self,
+        pre_commit_config: dict,
+        python_version: str,
+        native_images: bool,
+    ) -> None:
+        hooks = self.hooks(pre_commit_config)
+        PyclnHook(hooks["pycln"]).prepare_hook(PYTHONPATH)
+        RuffHook(hooks["ruff"]).prepare_hook(python_version, IS_GITHUB_ACTIONS)
+        MypyHook(hooks["mypy"]).prepare_hook(python_version)
+        RunUnitTestHook(hooks["run-unit-tests"]).prepare_hook(native_images)
+
+    def run(
+        self,
+        test: bool = False,
+        skip_hooks: Optional[List[str]] = None,
+        validate: bool = False,
+        format: bool = False,
+        native_images: bool = False,
+        verbose: bool = False,
+        show_diff_on_failure: bool = False,
+    ) -> int:
+        # handle skipped hooks
+        ret_val = 0
+        precommit_env = os.environ.copy()
+        skipped_hooks: set = SKIPPED_HOOKS
+        skipped_hooks |= set(skip_hooks or [])
+        if os.getenv("CI"):
+            # No reason to update the docker-image on CI, as we don't commit from the CI
+            skipped_hooks.add("update-docker-image")
+        if not test and not native_images:
+            skipped_hooks.add("run-unit-tests")
+        if validate:
+            skipped_hooks.remove("validate")
```

Comment:
what if "validate" is not in the set? won't we get `KeyError`?

Reply:
It shouldn't happen. Regardless I added a check for it